include/cocaine/api/isolate.hpp

```diff
@@ -67,8 +68,29 @@ struct isolate_t {
     }
 
     virtual
+    void
+    spool() = 0;
+
+    // Default implementation delegates the control flow into the blocking spool method.
+    virtual
     std::unique_ptr<cancellation_t>
-    spool(callback_type cb) = 0;
+    async_spool(callback_type cb) {
+        std::unique_ptr<cancellation_t> cancellation(new cancellation_t());
```

Comment:
What is the purpose of same empty cancellation_t for all? If you want exception safe spool I'd return cancellation_t from private pure virtual spool_impl method and make exception processing here in public spool method. Or if we don't use cancellation at all maybe it's better to drop it completely?